### PR TITLE
fix(gatsby-plugin-page-creator): juggle reporter dont depend on cli

### LINKED ISSUE #26345: Cannot find module 'gatsby-cli/lib/reporter'
After updating from gatsby 2.24.36  to version 2.24.40 getting error on npm start:
Error in "/Users/Austin/Projects/schwayweb/node_modules/gatsby-plugin-page-creator/gatsby-node.js": Cannot find module 'gatsby-cli/lib/reporter'

entire output here: https://gist.github.com/theskillwithin/2482854d8aa23e91885799925516c833

```

  System:
    OS: macOS 10.15.6
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.14.1 - ~/.nvm/versions/node/v12.14.1/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.6 - ~/.nvm/versions/node/v12.14.1/bin/npm
  Languages:
    Python: 2.7.17 - /usr/local/bin/python
  Browsers:
    Chrome: 84.0.4147.105
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    gatsby: ^2.24.40 => 2.24.40 
    gatsby-background-image: ^1.1.1 => 1.1.1 
    gatsby-image: ^2.4.15 => 2.4.15 
    gatsby-plugin-emotion: ^4.3.10 => 4.3.10 
    gatsby-plugin-google-analytics: ^2.3.13 => 2.3.13 
    gatsby-plugin-layout: ^1.3.10 => 1.3.10 
    gatsby-plugin-manifest: ^2.4.22 => 2.4.22 
    gatsby-plugin-offline: ^3.2.23 => 3.2.23 
    gatsby-plugin-react-helmet: ^3.3.10 => 3.3.10 
    gatsby-plugin-root-import: ^2.0.5 => 2.0.5 
    gatsby-plugin-sharp: ^2.6.26 => 2.6.26 
    gatsby-source-contentful: ^2.3.36 => 2.3.36 
    gatsby-source-filesystem: ^2.3.24 => 2.3.24 
    gatsby-source-shopify: ^3.2.26 => 3.2.26 
    gatsby-transformer-remark: ^2.8.28 => 2.8.28 
    gatsby-transformer-sharp: ^2.5.12 => 2.5.12 
```

### PR DESCRIPTION
**Pull request by pvdz**
Resolves an issue where something was imported from `gatsby-cli` without depending on it.

We'd rather not have a plugin depend on the cli directly and since the `reporter` is already passed on into it we'd rather juggle it forward.

Fixes #26345

### CODE REVIEW

File: packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import sysPath from "path"
-import reporter from "gatsby-cli/lib/reporter"
+import { Reporter } from "gatsby"
```
Comment (sidharthachatterjee) [inline packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts:2]: Should this be a type only import? 

```suggestion
import type { Reporter } from "gatsby"
```

File: packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { createClientOnlyPage } from "./create-client-only-page"
 import { createPagesFromCollectionBuilder } from "./create-pages-from-collection-builder"
 import systemPath from "path"
 import { trackFeatureIsUsed } from "gatsby-telemetry"
+import { Reporter } from "gatsby"
```
Comment (sidharthachatterjee) [inline packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts:7]: Should this be a type only import?

Comment (pvdz) [reply packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts:7]: As discussed, nope because we decided not to use `import type`.

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.24.42`

**Pull request merged**